test/unit/transitions/multi_form_alerts.js

```diff
@@ -0,0 +1,384 @@
+const _ = require('underscore'),
+      config = require('../../../config'),
+      lineage = require('../../../lib/lineage'),
+      messages = require('../../../lib/messages'),
+      sinon = require('sinon').sandbox.create(),
+      transition = require('../../../transitions/multi_form_alerts'),
+      utils = require('../../../lib/utils'),
+      vm = require('vm');
+
+exports.tearDown = callback => {
+  sinon.restore();
+  callback();
+};
+
+exports.setUp = callback => {
+  sinon.stub(utils, 'getReportsWithinTimeWindow').returns(Promise.resolve(reports));
+  sinon.stub(lineage, 'fetchHydratedDocPromise').callsFake(id => {
+    if (id === doc._id) { return Promise.resolve(doc); }
+    if (id === reports[0]._id) { return Promise.resolve(hydratedReports[0]); }
+    if (id === reports[1]._id) { return Promise.resolve(hydratedReports[1]); }
+  });
+  sinon.spy(vm, 'runInNewContext'); // spy, not stub, to pass through
+  sinon.stub(messages, 'addError');
+  sinon.stub(messages, 'addMessage');
+  callback();
+};
+
+const alert = {
+  isReportCounted: 'function() { return true; }',
+  numReportsThreshold : 3,
+  message : 'hi',
+  recipients : ['+254777888999'],
+  timeWindowInDays : 7
+};
+
+// doc is hydrated before being passed to the transition.
+const doc = {
+  _id: 'doc',
+  reported_date: 12345,
+  form: 'A',
+  contact: {
+    _id: 'contact',
+    phone: '+123456'
+  }
+};
+
+const reports = [
+  { _id: 'docA', form: 'A', contact: { _id: 'contactA' } },
+  { _id: 'docB', form: 'B', contact: { _id: 'contactB' } },
+];
+
+const hydratedReports = [
+  { _id: 'docA', form: 'A', contact: { _id: 'contactA', phone: '+234567'} },
+  { _id: 'docB', form: 'B', contact: { _id: 'contactB', phone: '+345678'} }
+];
+
+exports['function signature'] = test => {
+  test.ok(_.isFunction(transition.onMatch));
+  test.equal(transition.onMatch.length, 4);
+
+  test.ok(_.isFunction(transition.filter));
+  test.equal(transition.filter.length, 1);
+  test.done();
+};
+
+exports['filter validation'] = test => {
+  test.equal(transition.filter({}), false);
+  test.equal(transition.filter({
+    form: 'x',
+    type: 'badtype'
+  }), false);
+  test.equal(transition.filter({
+    type: 'data_record'
+  }), false);
+  test.equal(transition.filter({
+    form: 'x',
+  }), false);
+  test.equal(transition.filter({
+    form: 'x',
+    type: 'data_record'
+  }), true);
+  test.done();
+};
+
+exports['filter validation hasRun'] = test => {
+  test.equal(transition.filter({
+    form: 'x',
+    type: 'data_record',
+    transitions : { multi_form_alerts: 'hi' }
+  }), false);
+  test.done();
+};
+
+const testValidate = (test, alert) => {
```

Comment:
Maybe rename this to `testInvalidConfiguration` (or something) because it's hardcoded to only test validation failures.